src/use-swr.ts

```diff
@@ -45,22 +45,31 @@ const getGlobalState = (cache: Cache) => {
   ]
 }
 
-// Setup DOM events listeners for `focus` and `reconnect` actions
-if (!IS_SERVER) {
+function attachGlobalEvents(config: Configuration) {
   const [FOCUS_REVALIDATORS, RECONNECT_REVALIDATORS] = getGlobalState(
-    defaultConfig.cache
+    config.cache
   )
   const revalidate = (revalidators: Record<string, Revalidator[]>) => {
-    if (!defaultConfig.isDocumentVisible() || !defaultConfig.isOnline()) return
+    if (!config.isDocumentVisible() || !config.isOnline()) return
```

Comment:
If we don't have these 2 conditions, it will be easy to put `attachGlobalEvents` inside `createCache`...